src/Providers/Maps/Leaflet/Features/Shape.ts

```diff
@@ -0,0 +1,161 @@
+
+        public containsLocation(
+            mapId: string,
+            shapeId: string,
+            pointCoordinates: string,
+            coordinatesList: string
+        ): OSFramework.Maps.OSStructures.ReturnMessage {
+            this._map = OutSystems.Maps.MapAPI.MapManager.GetMapById(mapId);
+            // Set the default return message to prevent different else if's
+            this._returnObjSuccess = false;
+            this._returnObjCode =
+                OSFramework.Maps.Enum.ErrorCodes.CFG_InvalidMapId;
+
+            // Check if map exists
+            if (this._map) {
+                this._isInsideShape = false;
+                this._markerCoordinates = JSON.parse(
+                    pointCoordinates.toLocaleLowerCase()
+                );
+
+                // Check if shape exists
+                if (shapeId) {
+                    this._shape =
+                        OutSystems.Maps.MapAPI.ShapeManager.GetShapeById(
+                            shapeId
+                        );
+
+                    // Check if shape contains marker based on shape type
+                    switch (this._shape.type) {
+                        case OSFramework.Maps.Enum.ShapeType.Polyline:
+                            // The Polyline is an unsupported shape to use the ContainsLocation API
+                            this._returnObjCode =
+                                OSFramework.Maps.Enum.Unsupported.code;
+                            this._returnObjMessage =
+                                OSFramework.Maps.Enum.Unsupported.message;
+                            break;
+                        case OSFramework.Maps.Enum.ShapeType.Circle: {
+                            this._shapeCircle();
+                            break;
+                        }
+                        default: {
+                            // Default validations to check if marker is inside shape
+                            this._polyPoints =
+                                this._shape.provider.getLatLngs()[0];
+                            this._shapeDefault();
+                        }
+                    }
+                } else {
+                    const shapeCoordinatesList = JSON.parse(coordinatesList);
+                    this._polyPoints = [];
+
+                    // To create a shape we need at least 3 coordinates
+                    if (shapeCoordinatesList.length >= 3) {
+                        shapeCoordinatesList.forEach((item) => {
+                            this._polyPoints.push(L.latLng(item.Lat, item.Lng));
+                        });
+
+                        this._shapeDefault();
+                    }
+                }
+            }
+            // Set the return message to expose values
+            return {
+                isSuccess: this._returnObjSuccess,
+                code: this._returnObjCode,
+                message: this._returnObjMessage
+            };
+        }
+
+        public dispose(): void {
```

Comment:
We can delete this after delete the implementation.

Reply:
Done

Reply:
done